docs/FAQ.md

```diff
@@ -0,0 +1,236 @@
+
+For input fields, you often want to tell genqlient to send null to the server if the argument is set to the zero value, similar to the JSON `omitempty` tag.  In this case, you can do:
+
+```graphql
+query MyQuery(
+  # @genqlient(omitempty: true)
+  arg: String,
+) {
+  myField
+}
+```
+
+You can also put the `# @genqlient(omitempty: true)` on the first line, which will apply it to all arguments in the query.
+
+If you need to distinguish null from the empty string (or generally from the Go zero value of your type), you can tell genqlient to use a pointer for the field or argument like this:
+```graphql
+query MyQuery(
+  # @genqlient(pointer: true)
+  arg: String,
+) {
+  # @genqlient(pointer: true)
+  myField
+}
+```
+
+This will generate a Go field `MyField *string`, and set it to `nil` if the server returns null (and in reverse for arguments).  Such fields can be harder to work with in Go, but allow a clear distinction between null and the Go zero value.
+
+See [genqlient_directive.graphql](genqlient_directive.graphql) for complete documentation on these options.
+
+### … GraphQL interfaces?
+
+If you request an interface field, genqlient generates an interface type corresponding to the GraphQL interface, and several struct types corresponding to its implementations.  For example, given a query:
+
+```graphql
+query GetBooks {
+  favorite {
+    title
+    ... on Novel {
+      protagonist
+    }
+    ... on Dictionary {
+      language
+    }
+  }
+}
+```
+
+genqlient will generate the following types:
+
+```go
+type GetBooksFavoriteBook interface {
+  GetTitle() string
+}
+type GetBooksFavoriteNovel struct {
+  Title string
+  Protagonist string
+}
+type GetBooksFavoriteDictionary struct {
+  Title string
+  Language string
+}
+// (similarly for any other types that implement Book)
+```
+
+These can be used in the ordinary Go ways: to access shared fields, use the interface methods; to access type-specific fields, use a type switch:
+
+```go
+resp, err := GetBooks(...)
+fmt.Println("Favorite book:", resp.Favorite.GetTitle())
+if novel, ok := resp.Favorite.(*GetBooksFavoriteNovel); ok {
+  fmt.Println("Protagonist:", novel.Protagonist)
+}
+```
+
+### … documentation on the output types?
+
+For any GraphQL types or fields with documentation in the GraphQL schema, genqlient automatically includes that documentation in the generated code's GoDoc.  To add additional information to genqlient entrypoints, you can put comments in the GraphQL source:
+
+```graphql
+# This query gets the current user.
+#
+# @genqlient(...)
```

Comment:
what is supposed to go inside `@genqlient(...)`?

Reply:
Ah, this is just to show that if you have a `genqlient` directive, you can put it after the documentation.  I'll clarify.